---
title: MDSC: Trannscriptomic markers of MDSCs, neutrophils and monocytes
author: Slim Fourati
date: February 23, 2019
output: github_documents
---

loading require packages

```r
suppressPackageStartupMessages(library(package = "knitr"))
suppressPackageStartupMessages(library(package = "GEOquery"))
suppressPackageStartupMessages(library(package = "tidyverse"))

suppressPackageStartupMessages(library(package = "readxl"))
suppressPackageStartupMessages(library(package = "Biobase"))
suppressPackageStartupMessages(library(package = "limma"))
suppressPackageStartupMessages(library(package = "graphite"))
suppressPackageStartupMessages(library(package = "igraph"))
suppressPackageStartupMessages(library(package = "biomaRt"))
suppressPackageStartupMessages(library(package = "pheatmap"))
suppressPackageStartupMessages(library(package = "GSEABase"))
```

set default options/variables

```r
workDir <- dirname(getwd())
opts_chunk$set(tidy = FALSE, fig.path = "../figure/")
options(stringsAsFactors  = FALSE,
        readr.num_columns = 0,
	width             = 80)
# result will be written in directory called advanced
gseaDir <- file.path(workDir, "advanced")
if (!file.exists(gseaDir)) {
  flag <- dir.create(path = gseaDir)
}
```

Read Heim CE et al GMT
Reference: Heim CE 2018 Infect Immunol.
Description: Ly6G+Ly6C+CD11bhigh MDSC, Ly6G+Ly6C+CD11blow PMN and Ly6G-Ly6C+
	monocytes.

```r
gmtFile <- "HeimCE_2018_InfectImmun.SupplementaryFile.gmt"
cNames <- count_fields(file = file.path(workDir, "output", gmtFile),
		       tokenizer = tokenizer_tsv()) %>%
  max() %>%
  seq(from = 1) %>%
  paste0("X", .)
heimGMT <- read_tsv(file = file.path(workDir, "output", gmtFile),
		    col_names = cNames)
```


```r
# confirm genesets on geo dataset
countMat <- read_tsv(file =
                       file.path("/Users/sxf279/Desktop",
                                 "GSE118796_salmon_gene_counts.ensembl.txt.gz")) %>%
  as.data.frame() %>%
  mutate(ORF = gsub(pattern = "\\.[0-9]+$", replacement = "", ORF)) %>%
  column_to_rownames(var = "ORF")
normMat <- edgeR::cpm(countMat)
rowAnnotDF <- topDF %>%
  select(gene_id, modelName) %>%
  rename(geneset = modelName) %>%
  distinct() %>%
  column_to_rownames(var = "gene_id")
cat("heatmap of genes specific to mdsc, neutrophile or monocytes using top 200\n")
cat("differentially expressed genes (lr test p <= 0.05):")
pheatmap(normMat[topDF$gene_id, ],
         scale = "row",
         cluster_rows = FALSE,
         annotation_row = rowAnnotDF,
         cluster_cols = FALSE,
         cellwidth = 10,
	 show_rownames = FALSE)
```
Genes induced in MDSC compared to neutrophiles (PMN) are also induced when compared
to monocytes (MONO). They will be used as transcroptomic markers of MDSCs.

read MDSC frequencies

```r
inputFile <- file.path(dirname(workDir),
                       "Documents/Correspondances/20190125_V43DNA.emailMV",
                       "MDSCNEW paper.vSF.xlsx")
sheetLS <- c("DNA- all", "Ad26-all")
inputDF <- lapply(sheetLS, FUN = function(sheetName) {
  read_excel(path = inputFile, sheet = sheetName)
}) %>%
  do.call(what = rbind)
# add challenge                                                                  
challengeFile <- file.path(dirname(workDir),
                           "Documents/Correspondances/20170928_V43DNA.emailMV",
                           "V43 DNAAd26ALVAC MASTER DATA CLEAN.xls")
challengeDF <- read_excel(path = challengeFile,
                          sheet = "Animal info",
                          .name_repair = ~ make.names(., unique = TRUE))

inputDF <- merge(x = inputDF,
                 y = select(challengeDF, X, N.Challenges),
                 by.x = "AnimalID",
                 by.y = "X",
                 all.x = TRUE) %>%
  mutate(N.Challenges = gsub(pattern = "neg",
                               replacement = 11,
                               N.Challenges),
         N.Challenges = as.numeric(N.Challenges)) %>%
  gather(cname, value, -AnimalID, -Vaccine, -Timepoint, -N.Challenges) %>%
  mutate(cname = paste0(cname, ".", Timepoint)) %>%
  select(-Timepoint) %>%
  spread(cname, value)
```

Test if HLADR-CD14+ change over time

Frequencies of HLADR-CD14+ decreased significantly between prime and boost. There is
a non significant correlation between prime and boost HLADR-CD14+ frequencies. We
can't assume the HLADR-CD14+ frequencies measured at two differnt timepoints are
similar (ex. 2 weeks after 1st boost [week 14] and week 13 or 1 week after 2nd ALVAC
[week 25] and week 27).  
\hfill\break
\hfill\break
Check correlation of ARG1, CD66b with percentage of HLADR-CD14+ cells 

No significant correlation between ARG1 gene expression and percentage
of HLADR-/lo in CD15-CD14+. No probe match macaque CD66b.
\hfill\break
\hfill\break
Test association of MDSC genesets from the literature and percentage of HLADR-CD14+


\hfill\break
\hfill\break
Test association between MDSC markers and HLADR-CD14+ cells in V43

Transcriptomic markers of MDSCs as measured 2 weeks after 1st ALVAC boost are
positively correlated with HLADR-CD14+ measured by FCM.
\hfill\break
\hfill\break
Association between mdsc markers and challenge


Association between mdsc markers and stat3 pathway

```r
stat3DF <- read_csv(file = file.path(workDir, "code/stat3.slea.csv")) %>%
  rename(stat3.slea = slea)

plotDF <- sleaDF %>%
  rownames_to_column() %>%
  rename(mdsc.slea = slea) %>%
  merge(x = stat3DF, by.x = "Sample name", by.y = "rowname") %>%
  merge(y = pData(esetBaselined), by = "Sample name")

plotDF %>%
  group_by(prime, immunization, `time after immunization`) %>%
  do(rho = cor.test(formula = ~stat3.slea+mdsc.slea,
                    data = .,
                    method = "spearman")$estimate,
     p = cor.test(formula = ~stat3.slea+mdsc.slea,
                    data = .,
                    method = "spearman")$p.value) %>%
  mutate(rho = unlist(rho),
	 p = unlist(p))
```
MDSC markers are not significantly correlated with STAT3 pathway in the DNA
group.
\hfill\break
\hfill\break
Evaluate T-cell proliferation and T cell production of IFNG

Interferon-gamma response 24h after 2nd boost was negatively correlated to HLADR-CD14+
cells 2 weeks after prime. No assocatiation between T-cell proliferation genesets and
HLADR-CD14+
\hfill\break
\hfill\break
Test assocations of ROS, IDO, TGFB and PGE2 with HLADR-CD14+ cells

Several ROS related genesets are positively associated with HLADR-CD14+ measured 2
weeks after prime. TGFB signaling 24h after the 2nd ALVAC was negatively associated
with HLADR-CD14+. No association with PGE2 and HLADR-CD14+. No IDO genesets tested.
\hfill\break
\hfill\break

